src/Interpreters/HashJoin/HashJoin.cpp

```diff
@@ -340,22 +387,28 @@ size_t HashJoin::getTotalRowCount() const
 
 void HashJoin::doDebugAsserts() const
 {
-#ifndef NDEBUG
+#if !defined(NDEBUG) && 0
```

Comment:
How we maintain allocated size, which value we expect for each HashJoin instance now?

Reply:
i completely forgot about it, thanks for raising this topic.
blocks are shared now (for CHJ), but pair of (block, selector) is unique for each thread. so i thought we could just move this logic inside `ScatteredBlock` and calculate allocated bytes proportional to the number of rows that the given block owns. it seems fine for this calculations to be approximated, because we only use them to check if we exceeded some of the thresholds for memory usage.